web/frontend/src/pages/ballot/Show.tsx

```diff
@@ -48,108 +66,139 @@ const Ballot: FC = () => {
     }
   }, [postError, t]);
 
-  const handleCheck = (e: React.ChangeEvent<HTMLInputElement>) => {
-    setChoice(e.target.value);
-  };
-
   const hexToBytes = (hex: string) => {
-    for (let c = 0; c < hex.length; c += 2) bytes.push(parseInt(hex.substr(c, 2), 16));
+    const bytes: number[] = [];
+    for (let c = 0; c < hex.length; c += 2) {
+      bytes.push(parseInt(hex.substr(c, 2), 16));
+    }
     return new Uint8Array(bytes);
   };
 
-  const createBallot = (K: Buffer, C: Buffer) => {
-    let vote = JSON.stringify({ K: Array.from(K), C: Array.from(C) });
+  const createBallot = (EGPairs: Array<Buffer[]>) => {
+    const vote = [];
+    EGPairs.forEach(([K, C]) => vote.push({ K: Array.from(K), C: Array.from(C) }));
     return {
       ElectionID: electionID,
       UserId: sessionStorage.getItem('id'),
-      Ballot: Buffer.from(vote),
-      Token: token,
+      Ballot: vote,
     };
   };
 
   const sendBallot = async () => {
-    const [K, C] = encryptVote(choice, Buffer.from(hexToBytes(pubKey).buffer), edCurve);
-
+    const ballotChunks = voteEncode(answers, ballotSize, chunksPerBallot);
+    const EGPairs = Array<Buffer[]>();
+    ballotChunks.forEach((chunk) =>
+      EGPairs.push(encryptVote(chunk, Buffer.from(hexToBytes(pubKey).buffer), edCurve))
+    );
     //sending the ballot to evoting server
-    let ballot = createBallot(K, C);
-    let newRequest = {
+    const ballot = createBallot(EGPairs);
+    const newRequest = {
       method: 'POST',
       body: JSON.stringify(ballot),
     };
     setPostRequest(newRequest);
   };
 
   const handleClick = () => {
-    if (choice === '') {
-      userErrors.noCandidate = t('noCandidate');
-      setUserErrors(userErrors);
-      return;
+    if (ballotIsValid(configuration, answers, setAnswers)) {
+      setUserErrors('');
+      sendBallot();
+    } else {
+      setUserErrors(t('incompleteBallot'));
     }
-    sendBallot();
-    userErrors.noCandidate = '';
-    setUserErrors(userErrors);
   };
 
-  const electionClosedDisplay = () => {
-    return <div> {t('voteImpossible')}</div>;
+  const SubjectElementDisplay = (element: SubjectElement) => {
+    return (
+      <div>
+        {element.Type === RANK ? <Rank rank={element as RankQuestion} answers={answers} /> : null}
+        {element.Type === SELECT ? (
+          <Select select={element as SelectQuestion} answers={answers} setAnswers={setAnswers} />
+        ) : null}
+        {element.Type === TEXT ? (
+          <Text text={element as TextQuestion} answers={answers} setAnswers={setAnswers} />
+        ) : null}
+      </div>
+    );
   };
 
-  const possibleChoice = (candidate) => {
+  const SubjectTree = (subjectElement: SubjectElement) => {
+    const subject = subjectElement as Subject;
```

Comment:
Since subject.Elements contains SubjectElements, and SubjectTree is called recursively on them I thought it made sense, but yes it's better to actually only do a cast if the SubjectElements is a Subject !